### ISSUE TITLE
The training mode is accidentally enabled in training_step function.

### ISSUE THREAD

**Issue by w2kun** [PyTorchLightning/pytorch-lightning#18781]
### Bug description

I defined a neural network in the __init__ method of a lightning module, and some modules of this network are pre-trained. I fixed these pretrained modules and set their training mode to false. However, when training started, I found that the training mode of these modules was unexpectedly set to true. I think this may be a potential bug.

### What version are you seeing the problem on?

v1.9

### Error messages and logs

```
# Error messages and logs here please
```

### Environment

<details>
  <summary>Current environment</summary>

```
#- Lightning Component (e.g. Trainer, LightningModule, LightningApp, LightningWork, LightningFlow):
#- PyTorch Lightning Version (e.g., 1.5.0):
#- Lightning App Version (e.g., 0.5.2):
#- PyTorch Version (e.g., 2.0):
#- Python version (e.g., 3.9):
#- OS (e.g., Linux):
#- CUDA/cuDNN version:
#- GPU models and configuration:
#- How you installed Lightning(`conda`, `pip`, source):
#- Running environment of LightningApp (e.g. local, cloud):
```

</details>
